src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleDictionary.cs

```diff
@@ -53,9 +53,15 @@ private static void HandleStartDictionary(JsonSerializerOptions options, ref Rea
 
                     state.Current.ReturnValue = classInfo.CreateObject();
                 }
-                else
+                else if (state.Current.IsProcessingObject(ClassType.Enumerable))
                 {
-                    ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(classInfo.Type);
+                    // Array with metadata within the dictionary.
+                    HandleStartObjectInEnumerable(ref state, options, classInfo.Type);
+
+                    Debug.Assert(options.ReferenceHandling.ShouldReadPreservedReferences());
+                    Debug.Assert(state.Current.JsonClassInfo!.Type.GetGenericTypeDefinition() == typeof(JsonPreservedReference<>));
+
+                    state.Current.ReturnValue = state.Current.JsonClassInfo.CreateObject!();
```

Comment:
We only get in this code path if `IsProcessingDictionary() == true` which is not the case for `JsonPreservableArrayReference<T>` objects.

Since the enumerables are treated as `ClassType.Object` when they were wrapped in `JsonPreservableArrayReference<T>` we will only get nested `JsonPreservableArrayReference<T>`s on `HandleStartObject()`.